Tests/SonarScanner.MSBuild.PreProcessor.Test/Infrastructure/MockIDownloaderHelper.cs

```diff
@@ -35,14 +38,84 @@ public static Mock<IDownloader> CreateMock()
         public static Mock<IDownloader> SetupTryDownloadIfExists(this Mock<IDownloader> mock, Uri requestUri = null, string response = null)
         {
             mock.Setup(x => x.TryDownloadIfExists(It.Is<Uri>(downloadUri => requestUri == null || downloadUri == requestUri), It.IsAny<bool>()))
-                .ReturnsAsync(Tuple.Create(response != null, response));
+                .ReturnsAsync(Tuple.Create(response != null, response))
```

Comment:
Thinking more about this helper, I feel that this got a bit out of hand. I'm not sure if it was a good idea.

- we always setup dispose although it's not always needed
- all the setups are marked as verifiable although not always needed
- we have conditions like `requestUri == null || downloadUri == requestUri` in order to be generic but this will not always match the needs of the tests (we might want to be specific with parameters)
- there is plenty of duplicated code that defeats the initial purpose
- the name itself contains `Helper` which is a code smell

I'm thinking now that maybe we could either inline most of the conditions and remove the helper, or make it a factory that can be used like `factory.WithGetBaseUri().WithDownload().Create()` inclining more to inline the code.

@sebastien-marichal, @gregory-paidis-sonarsource: what do you think?

Reply:
As discussed @costin-zaharia-sonarsource, I completely removed the `MockIDownloaderHelper` class and inlined all the existing mock